### PR TITLE
Management documentation updates

### LINKED ISSUE #2146: Documentation for generating a SAS token/url for ingress/egress
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates
- [x] This is a request for a change to documentation
- [x] This isn't a feature request or bug report
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :ledger: Where is the problem?

The security checklist directs readers to an administrator document in several places (e.g. https://data-safe-haven--2141.org.readthedocs.build/en/2141/deployment/security_checklist.html#check-that-the-role-system-manager-can-send-an-upload-token-to-the-role-data-provider-representative-over-a-secure-channel) that doesn't exist. 

## :memo: Required documentation changes

Add some documentation explaining what permissions access tokens require etc

### LINKED ISSUE #2252: Enhance the teardown documentation for clarity
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates
- [x] This is a request for a change to documentation
- [x] This isn't a feature request or bug report
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :ledger: Where is the problem?

https://data-safe-haven.readthedocs.io/en/develop/management/index.html#removing-a-deployed-data-safe-haven

## :memo: Required documentation changes

- Explain a bit more exactly what will happen when running `dsh sre teardown YOUR_SRE_NAME` or `dsh shm teardown` so System Managers new to DSH don't delete things without meaning
- Add some big scary warnings, especially to the SHM teardown part saying "deleting this will affect all SREs" etc
- I'm not sure if in DSH v5 the storage account is now falling under the SRE resources as oppose to the SHM subscription (as was the case in v4)? If so, also add big scary warning saying data will be deleted by SRE teardown (unless this is not the case, in which case explain that)

### PR DESCRIPTION
**Pull request by craddm**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :arrow_heading_up: Summary

Updates various documentation sections to provide additional detail.

Specifically, it expands upon data egress/ingress procedures and tearing down of SHMs and SREs.

### :closed_umbrella: Related issues

Contributes to #2252 
Closes #2146 

### :microscope: Tests

Documentation only

### CODE REVIEW

File: docs/source/management/index.md [hunk 1]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
 ```{code} shell
 $ dsh sre teardown YOUR_SRE_NAME
 ```
 
+::::{admonition} Tearing down an SRE deletes ALL data and resources
+Running `dsh sre teardown` will remove all resources deployed within the SRE.
+Ensure that any desired outputs have been egressed before deleting the SRE.
+Any data remaining on the SRE will be deleted.
+The user groups for the SRE on Microsoft Entra ID will also be deleted.
+::::
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
::::{danger} Tearing down an SRE is destructive and irreversible 
Running `dsh sre teardown` will destroy **all** resources deployed within the SRE.
Ensure that any desired outputs have been extracted before deleting the SRE.
**All** data remaining on the SRE will be deleted.
The user groups for the SRE on Microsoft Entra ID will also be deleted.
::::
```

Some suggestions for changing the wording and emphasis. (And the classic, egress is not a verb).

File: docs/source/management/index.md [hunk 2]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
 - Run the following if you want to teardown the deployed SHM:
 
 ```{code} shell
 $ dsh shm teardown
 ```
+
+::::{admonition} Tearing down an SHM
```
Comment (JimMadge) [inline docs/source/management/index.md:137]:
```suggestion
::::{warning} Tearing down an SHM
```

File: docs/source/management/index.md [hunk 3]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+All SREs associated with the SHM must be torn down before the SHM can be torn down.
```
Comment (JimMadge) [inline docs/source/management/index.md]: Is this true? As in, if I run `dsh shm teardown` now will it stop me if there are SREs?

File: docs/source/management/index.md [hunk 4]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+Tearing down the SHM deletes any remotely stored SRE configurations.
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
Tearing down the SHM permanently deletes **all** remotely stored configuration and state data.
```

File: docs/source/roles/data_provider_representative/data_ingress.md [hunk 5]
```diff
@@ -5,9 +5,20 @@
...
+Talk to your {ref}`role_system_manager` to discuss possible methods of bringing data into the environments.
+It may be convenient to use [Azure Storage Explorer](https://azure.microsoft.com/en-us/products/storage/storage-explorer/).
+In this case you will not need log-in credentials, as your {ref}`role_system_manager` can provide a short-lived secure access token which will let you upload data.
+
+```{tip}
+You may want to keep the following considerations in mind when transferring data in order to reduce the chance of a data breach
+- use of short-lived access tokens limits the time within which an attacker can operate
+- letting your {ref}`role_system_manager` know a fixed IP address you will be connecting from (eg. a corporate VPN) limits the places an attacker can operate from
+- communicating with your {ref}`role_system_manager` through a secure out-of-band channel (eg. encrypted email) reduces the chances that an attacker can intercept or alter your messages in transit
+```
```
Comment (JimMadge) [inline docs/source/roles/data_provider_representative/data_ingress.md]:
````suggestion
```{tip}
You may want to keep the following considerations in mind when transferring data in order to reduce the chance of a data breach.

- Use of short-lived access tokens limits the time within which an attacker can operate.
- Letting your {ref}`role_system_manager` know a fixed IP address you will be connecting from (_e.g._ a corporate VPN) limits the places an attacker can operate from.
- Communicating with your {ref}`role_system_manager` through a secure out-of-band channel (_e.g._ encrypted email) reduces the chances that an attacker can intercept or alter your messages in transit.

```
````

File: docs/source/management/index.md [hunk 6]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+::::
+
+## Managing data ingress and egress
+
+### Data Ingress
+
+It is the data provider's responsibility to upload the data required by the safe haven.
```
Comment (JimMadge) [inline docs/source/management/index.md]: I think this isn't true. Isn't this what we say the DPRs role is?

File: docs/source/management/index.md [hunk 7]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
+```{important}
+Any data ingress must be signed off by the {ref}`role_data_provider_representative`, {ref}`role_investigator` and referee (if applicable).
+```
```
Comment (JimMadge) [inline docs/source/management/index.md]: This feels out of scope.

File: docs/source/management/index.md [hunk 8]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
+The following steps show how to generate a temporary write-only upload token that can be securely sent to the data provider, enabling them to upload the data:
+
+- In the Azure portal select `Subscriptions` then navigate to the subscription containing the relevant SHM
+- Search for the resource group: `shm-<SHM ID>-sre-<SRE ID>-rg`, then click through to the storage account called: `sh<first three letter of SHM ID>sre<first three letters of SRE ID>sensitivedata`
+- Click `Networking` under `Settings` and ensure that the data provider's IP address is one of those allowed under the `Firewall` header, then hit the save icon in the top left
```
Comment (JimMadge) [inline docs/source/management/index.md]: I don't think we use the terms `SHM ID` and `SRE ID` any more.

Comment (JimMadge) [inline docs/source/management/index.md]: We have been using the [`guilabel` role](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-guilabel) for buttons in the documentation.

For example,

https://github.com/alan-turing-institute/data-safe-haven/blob/d2141434b496c5863ff8065537ebf56cbd78ef62/docs/source/deployment/configure_entra_id.md?plain=1#L45-L50

File: docs/source/management/index.md [hunk 9]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+    - Set a 24 hour time window in the `Start and expiry date/time` (or an appropriate length of time)
+    - Leave everything else as default and click `Generate SAS token and URL`
+    - Copy the `Blob SAS URL`
+
+      ```{image} ingress_token_write_only.png
+      :alt: write-only SAS token
+      :align: center
+      ```
+
+- Send the `Blob SAS URL` to the data provider through a secure channel (for example, you could use the [Egress secure email](https://www.egress.com/) service)
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
- Send the `Blob SAS URL` to the {ref}`role_data_provider_representative` through a secure channel
```

Probably should be careful not to promote a particular service.

File: docs/source/management/index.md [hunk 10]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+- The data provider should now be able to upload data
+- You can validate successful data ingress by logging into a workspace in the SRE and checking the `/mnt/input` volume, where you should be able to view the data that the data provider has uploaded
```
Comment (JimMadge) [inline docs/source/management/index.md]: These are instructions for a manager? I don't think we would want them to login to the SRE.

File: docs/source/management/index.md [hunk 11]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
+### Data egress
+
+```{important}
+Any data egress must be signed off by the {ref}`role_data_provider_representative`, {ref}`role_investigator` and referee (if applicable).
+```
```
Comment (JimMadge) [inline docs/source/management/index.md]: Again, this feels out of scope.

File: docs/source/management/index.md [hunk 12]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
+```{important}
+Assessment of output must be completed **before** an egress link is created.
```
Comment (JimMadge) [inline docs/source/management/index.md:181]: I think we probably can say this. It might help to say why though.

File: docs/source/management/index.md [hunk 13]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+```
+
+The {ref}`role_system_manager` creates a time-limited and IP restricted link to remove data from the environment, after the outputs have been classified and approved for release.
```
Comment (JimMadge) [inline docs/source/management/index.md]: Classification feels out of scope.

File: docs/source/management/index.md [hunk 14]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
+- In the Azure portal select `Subscriptions` then navigate to the subscription containing the relevant SHM
+- Search for the resource group: `shm-<SHM ID>-sre-<SRE ID>-rg`, then click through to the storage account called: `sh<first three letter of SHM ID>sre<first three letters of SRE ID>sensitivedata`
+- Click `Networking` under `Settings` to check the list of pre-approved IP addresses allowed under the `Firewall` header
+    - Ensure that the IP address of the person to receive the outputs is listed and enter it if not
```
Comment (JimMadge) [inline docs/source/management/index.md]: Do we want to encourage people to make ad-hoc changes like this, or use the CLI?

File: docs/source/management/index.md [hunk 15]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+    - Set a time window in the `Start and expiry date/time` that gives enough time for the person who will perform the secure egress download to do so
+    - Leave everything else as default click `Generate SAS token and URL`
+    - Copy the `Blob SAS URL`
+
+      ```{image} egress_token_read_only.png
+      :alt: Read-only SAS token
+      :align: center
+      ```
+
+- Send the `Blob SAS URL` to the relevant person through a secure channel (for example, you could use the [Egress secure email](https://www.egress.com/) service)
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
- Send the `Blob SAS URL` to the relevant person through a secure channel
```

File: docs/source/management/index.md [hunk 16]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+- The appropriate person should now be able to download data
+
+### The output volume
+
+Once you have set up the egress connection in `Azure Storage Explorer`, you should be able to view data from the **output volume**, a read-write area intended for the extraction of results, such as figures for publication.
+On the workspaces, this volume is `/mnt/output` and is shared between all workspaces in an SRE.
+For more info on shared SRE storage volumes, consult the {ref}`Safe Haven User Guide <role_researcher_shared_storage>`.
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
For more information on shared SRE storage volumes, consult the {ref}`Safe Haven User Guide <role_researcher_shared_storage>`.
```

**Review by JimMadge — CHANGES_REQUESTED** [review]
This is looking good.

I think there are some changes we should make to keep it within scope and a few corrections.

File: docs/source/management/index.md (hunk 3 above)
Comment (craddm) [reply docs/source/management/index.md]: Depends if we want to merge #2256 (if I finish it in time)

File: docs/source/management/index.md (hunk 11 above)
Comment (craddm) [reply docs/source/management/index.md]: Egress in general?

File: docs/source/management/index.md (hunk 12 above)
Comment (craddm) [reply docs/source/management/index.md:181]: Will just remove if we're not talking about classification etc

File: docs/source/management/index.md (hunk 3 above)
Comment (JimMadge) [reply docs/source/management/index.md]: Let's bump that to 5.0.2

File: docs/source/management/index.md (hunk 11 above)
Comment (JimMadge) [reply docs/source/management/index.md]: The part I don't like is "must be signed off by ..." because that is imposing some set of IG processes. I don't think this documentation is the right place to enforce or suggest what that process is.

I do think it is worth having a prominent warning that outputs may be sensitive and so you need to take care.

File: docs/source/management/index.md (hunk 12 above)
Comment (JimMadge) [reply docs/source/management/index.md:181]: I think it would be good to keep. I can't imagine a functional TRE where there isn't some sort of output checking.

It's just that in these docs we need to talk about how DSH works and how to use it, not enforce a particular IG process. So I think the important point here to to highlight that outputs may be sensitive so you must take care, consider the outputs and have _some_ process.

I think talking about classification isn't possible because we either leave it undefined ("You must do classification", but what does classification mean?) or we define a classification process, which is definitely out of scope.

File: docs/source/management/index.md [hunk 17]
```diff
@@ -133,83 +133,79 @@ The user groups for the SRE on Microsoft Entra ID will also be deleted.
...
 ## Managing data ingress and egress
 
 ### Data Ingress
 
-It is the data provider's responsibility to upload the data required by the safe haven.
-
-```{important}
-Any data ingress must be signed off by the {ref}`role_data_provider_representative`, {ref}`role_investigator` and referee (if applicable).
-```
+It is the data provider representative's responsibility to upload the data required by the safe haven.
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
It is the {ref}`role_data_provider_representative`'s responsibility to upload the data required by the safe haven.
```

File: docs/source/management/index.md [hunk 18]
```diff
@@ -133,83 +133,79 @@ The user groups for the SRE on Microsoft Entra ID will also be deleted.
...
-- Click `Shared access tokens` under `Settings` and do the following:
-    - Under `Signing method`, select `User delegation key`
-    - Under `Permissions`, check these boxes:
-        - `Write`
-        - `List`
-    - Set a 24 hour time window in the `Start and expiry date/time` (or an appropriate length of time)
-    - Leave everything else as default and click `Generate SAS token and URL`
-    - Copy the `Blob SAS URL`
+- In the Azure portal select **Subscriptions** then navigate to the subscription containing the relevant SHM
+- Search for the resource group: **`shm-<YOUR_SHM_NAME>-sre-<YOUR_SRE_NAME>-rg`**, then click through to the storage account called: **`sh<first three letters of SHM name>sre<first three letters of SRE name>sensitivedata`**
```
Comment (JimMadge) [inline docs/source/management/index.md]: The "first three letters" thing is a bit inelegant. I'm not sure if that name is also garunteed. It might be better to just say the storage account ending with "sensitivedata"?

Comment (craddm) [reply docs/source/management/index.md]: Agreed, and oh, of course - that's much better.

File: docs/source/management/index.md [hunk 19]
```diff
@@ -27,7 +29,7 @@ Grace;Hopper;+18005550100;grace@nasa.gov;US
 $ dsh users add PATH_TO_MY_CSV_FILE
 ```
 
-## Listing available users
+### Listing available users
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
### List available users
```

File: docs/source/management/index.md [hunk 20]
```diff
@@ -87,7 +89,9 @@ If you have manually created a user and want to enable SSPR, do the following
         - **Email:** enter the user's email address here
     - Click the **{guilabel}`Save`** icon in the top panel
 
-## Listing available SRE configurations and deployment status
+## Managing SREs
+
+### Listing available SRE configurations and deployment status
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
### List available SRE configurations and deployment status
```

File: docs/source/management/index.md [hunk 21]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
 └──────────────┴──────────┘
 ```
 
-## Removing a deployed Data Safe Haven
+### Removing a deployed Data Safe Haven
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
### Remove a deployed Data Safe Haven
```

File: docs/source/management/index.md [hunk 22]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+All SREs associated with the SHM should be torn down before the SHM is torn down.
+::::
+
+## Managing data ingress and egress
+
+### Data Ingress
+
+It is the {ref}`role_data_provider_representative`'s responsibility to upload the data required by the safe haven.
+
+The following steps show how to generate a temporary write-only upload token that can be securely sent to the data provider, enabling them to upload the data:
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
The following steps show how to generate a temporary, write-only upload token that can be securely sent to the {ref}`role_data_provider_representative`, enabling them to upload the data:
```

File: docs/source/management/index.md [hunk 23]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
+- In the Azure portal select **Subscriptions** then navigate to the subscription containing the relevant SHM
+- Search for the resource group: **`shm-<YOUR_SHM_NAME>-sre-<YOUR_SRE_NAME>-rg`**, then click through to the storage account ending with **`sensitivedata`**
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
- Search for the resource group: `shm-<YOUR_SHM_NAME>-sre-<YOUR_SRE_NAME>-rg`, then click through to the storage account ending with `sensitivedata`
```

File: docs/source/management/index.md [hunk 24]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+
+```{important}
+Assessment of output must be completed **before** an egress link is created.
+Outputs are potentially sensitive, and so an appropriate process must be applied to ensure that they are suitable for egress.
+```
+
+The {ref}`role_system_manager` creates a time-limited and IP restricted link to remove data from the environment.
+
+- In the Azure portal select **Subscriptions** then navigate to the subscription containing the relevant SHM
+- Search for the resource group: **`shm-<YOUR_SHM_NAME>-sre-<YOUR_SRE_NAME>-rg`**, then click through to the storage account called: **`sh<first three letters of SHM name>sre<first three letters of SRE name>sensitivedata`**
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
- Search for the resource group: `shm-<YOUR_SHM_NAME>-sre-<YOUR_SRE_NAME>-rg`, then click through to the storage account ending with `sensitivedata`
```

File: docs/source/management/index.md [hunk 25]
```diff
@@ -108,16 +112,104 @@ Available SRE configurations for context 'green':
...
+      :alt: Read-only SAS token
+      :align: center
+      ```
+
+- Send the **Blob SAS URL** to the relevant person through a secure channel
+- The appropriate person should now be able to download data
+
+### The output volume
+
+Once you have set up the egress connection in `Azure Storage Explorer`, you should be able to view data from the **output volume**, a read-write area intended for the extraction of results, such as figures for publication.
```
Comment (JimMadge) [inline docs/source/management/index.md]:
```suggestion
Once you have set up the egress connection in Azure Storage Explorer, you should be able to view data from the **output volume**, a read-write area intended for the extraction of results, such as figures for publication.
```

**Review by JimMadge — CHANGES_REQUESTED** [review]
A few small suggestions but I think this is ready :+1:

**Review by JimMadge — APPROVED** [review]
LGTM :+1:.
